src/audio_pixel.h

```diff
@@ -37,45 +44,76 @@ class audio_pixel_t {
 class audio_pixel_block_t {
 
 public: 
-    // TODO: handle the accessor not being valid
-    audio_pixel_block_t(MediaItem_Take* take, double pix_per_s) 
-        : m_take(take), m_pix_per_s(pix_per_s), 
-          m_accessor(safe_audio_accessor_t(take)) 
-          { /* TODO: */}
-    ~audio_pixel_block_t() {}
-
-    void update() { /* TODO: see AudioAccessorStateChanged */}
-    bool flush() const { /*TODO*/ }; // flush contents to file
-
-    const std::vector<audio_pixel_t>& get_pixels() const { return m_pixels; }
-    const std::vector<audio_pixel_t>& get_pixels(double t0, double t1) const { /* TODO */}
+    audio_pixel_block_t(double pix_per_s)
+        :m_pix_per_s(pix_per_s) {};
+    audio_pixel_block_t(double pix_per_s, std::vector<std::vector<audio_pixel_t>> channel_pixels)
+        :m_pix_per_s(pix_per_s), m_channel_pixels(channel_pixels) {};
+    ~audio_pixel_block_t() {};
 
+    bool flush() const { /*TODO*/ }; // flush contents to file
     // serialization and deserialization
     void to_json(json& j) const;
     void from_json(const json& j);
 
+    const int get_mono_pixel_count() {
+        if (m_channel_pixels.size() > 0)
+            return m_channel_pixels[0].size();
+        return 0;
+    }
+    const int get_stero_pixel_count() {
+        if (m_channel_pixels.size() > 0)
+            return m_channel_pixels[0].size() * m_channel_pixels.size();
+        return 0;
+    }
+    const std::vector<std::vector<audio_pixel_t>>& get_pixels() const { return m_channel_pixels; }
+    const std::vector<std::vector<audio_pixel_t>>& get_pixels(double t0, double t1) const { /* TODO */}
+    double get_pps() { return m_pix_per_s; }
+
+
+    void update(std::vector<double>& sample_buffer, int num_channels, int sample_rate);
+
+
+
 private: 
-    MediaItem_Take *m_take {nullptr}; // parent take
+    std::vector<std::vector<audio_pixel_t>> m_channel_pixels; // vector of audio_pixels for each channel 
```

Comment:
maybe a type alias is needed for this 2d vector of audio pixels? `multichannel_audio_pixel_array_t` or `audio_pixel_array_2d_t` or something like that. 

for brevity, these should just be called `m_pixels`, but we should document that it is a 2d thing in (channels, samples) form.

Reply:
i gotcha, I started this but was having a few issues

Reply:
I'm gonna need some help on this issue due to how the types break down (for cases of const, &, *, etc...)